src/main/java/org/jabref/gui/dialogs/CaptchaSolverDialog.java

```diff
@@ -0,0 +1,56 @@
+package org.jabref.gui.dialogs;
+
+import java.util.concurrent.CountDownLatch;
+
+import javafx.application.Platform;
+import javafx.scene.control.ButtonType;
+import javafx.scene.web.WebView;
+
+import org.jabref.gui.util.BaseDialog;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.net.URLDownload;
+
+import org.jsoup.helper.W3CDom;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+
+public class CaptchaSolverDialog extends BaseDialog<String> implements org.jabref.logic.importer.fetcher.CaptchaSolver {
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(CaptchaSolverDialog.class);
+
+    private WebView webView;
+
+    public CaptchaSolverDialog() {
+        super();
+        this.setTitle(Localization.lang("Captcha Solver"));
+        getDialogPane().getButtonTypes().add(ButtonType.CLOSE);
+        getDialogPane().lookupButton(ButtonType.CLOSE).setVisible(true);
+
+        webView = new WebView();
+        webView.getEngine().setJavaScriptEnabled(true);
+        webView.getEngine().setUserAgent(URLDownload.USER_AGENT);
+        getDialogPane().setContent(webView);
+    }
+
+    @Override
+    public String solve(String queryURL) {
+        // slim implementation of https://news.kynosarges.org/2014/05/01/simulating-platform-runandwait/
+        final CountDownLatch doneLatch = new CountDownLatch(1);
```

Comment:
You need to listen for the web engine ready event, see the preview Tab viewer where we add this highlight ja stuff

Reply:
```
   previewView.getEngine().getLoadWorker().stateProperty().addListener((observable, oldValue, newValue) -> {

            if (newValue != Worker.State.SUCCEEDED) {
                return;
            }
```
See https://openjfx.io/javadoc/11/javafx.web/javafx/scene/web/WebEngine.html

Reply:
> You need to listen for the web engine ready event, see the preview Tab viewer where we add this highlight ja stuff 

Is this happen synchronously? The interface for the Captcha solver is designed in a synchronous way. Otherwise all fetchers need to be changed.

I'll be away anyway for the next days. Thus, you are free to experiment 😅